packages/expo/src/ScreenOrientation/ExpoScreenOrientation.web.ts

```diff
@@ -1,8 +1,100 @@
+import { SyntheticPlatformEmitter } from 'expo-core';
+
+import { OrientationInfo, OrientationLock } from './ScreenOrientation.types';
+
+const WebOrientationLock = {
+  any: 'ALL',
+  portrait: 'PORTRAIT',
+  'portrait-primary': 'PORTRAIT_UP',
+  'portrait-secondary': 'PORTRAIT_DOWN',
+  landscape: 'LANDSCAPE',
+  'landscape-primary': 'LANDSCAPE_LEFT',
+  'landscape-secondary': 'LANDSCAPE_RIGHT',
+  natural: 'ALL_BUT_UPSIDE_DOWN',
+};
+
+const OrientationLockJSONToNative = {
+  ALL: 'any',
+  PORTRAIT: 'portrait',
+  PORTRAIT_UP: 'portrait-primary',
+  PORTRAIT_DOWN: 'portrait-secondary',
+  LANDSCAPE: 'landscape',
+  LANDSCAPE_LEFT: 'landscape-primary',
+  LANDSCAPE_RIGHT: 'landscape-secondary',
+  ALL_BUT_UPSIDE_DOWN: 'natural',
+};
+
+const OrientationAngleJSONToNative = {
+  '0': 'portrait-primary',
+  '180': 'portrait-secondary',
+  '-180': 'portrait-secondary',
+  '90': 'landscape-primary',
+  '-90': 'landscape-secondary',
+  '270': 'landscape-secondary',
+};
+
+declare const window: Window;
+
+const { screen } = window;
+
+const OrientationTarget: ScreenOrientation =
+  screen['msOrientation'] || (screen.orientation || screen['mozOrientation']);
+
+function emitOrientationEvent() {
+  SyntheticPlatformEmitter.emit('didUpdateDimensions', {});
```

Comment:
in the android and web layer, the event is simply triggered, and the orientation/orientationlock is determined when it handles the event

Reply:
hm. Since the `didUpdateDimensions` event contains a bunch of information that isnt pertinent to screen orientation, i'll change it to emit an `expoDidUpdateDimensions` event instead